src/third_party/minitrace.h

```diff
@@ -139,8 +139,8 @@ void internal_mtr_raw_event_arg(const char *category, const char *name, char ph,
 
 // Metadata. Call at the start preferably. Must be const strings.
 
-#define MTR_META_PROCESS_NAME(n) internal_mtr_raw_event_arg("", "process_name", 'M', 0, MTR_ARG_TYPE_STRING_COPY, "name", (void *)(n))
-#define MTR_META_THREAD_NAME(n) internal_mtr_raw_event_arg("", "thread_name", 'M', 0, MTR_ARG_TYPE_STRING_COPY, "name", (void *)(n))
+#define MTR_META_PROCESS_NAME(n) internal_mtr_raw_event_arg("", "process_name", 'M', 0, MTR_ARG_TYPE_STRING_CONST, "name", (void *)(n))
+#define MTR_META_THREAD_NAME(n) internal_mtr_raw_event_arg("", "thread_name", 'M', 0, MTR_ARG_TYPE_STRING_CONST, "name", (void *)(n))
```

Comment:
Could you please explain this part of the fix? Why doesn't it work to copy the process string?

I would prefer not to modify third party source code if possible.

Reply:
Not sure, actually. It used to work before (C), but it seems the copy happens "too late" in the process (C++).
My initial feeling was that it should be totally the opposite, "copy" would be safe and "const" dangerous...

```C
        case MTR_ARG_TYPE_STRING_CONST: ev->a_str = (const char*)arg_value; break;
        case MTR_ARG_TYPE_STRING_COPY: ev->a_str = strdup((const char*)arg_value); break;
```

It's like those "strdup" strings are deallocated as part of the shutdown process, before destructor is called.
Maybe there is a way to deallocate it more explicitly, that would avoid having to patch third party code.

Or do some other fix, that could go upstream.

Reply:
Adding an explicit `mtr_ flush` didn't help at all, and valgrind says that both strings are leaked (never freed).

So I don't know what could cause `strlen` to segfault like that, when called on a string literal or a `c_str()` ?